File: task3/Javascript/Subtask3.js
const readline = require('readline');

// Create an interface to read input from the console
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Function to generate the diamond pattern
function printDiamond(n) {
    let pattern = '';

    // Generate the upper part of the diamond
    for (let i = 0; i < n; i++) {
        pattern += ' '.repeat(n - i - 1) + '*'.repeat(2 * i + 1) + '\n';
    }

    // Generate the lower part of the diamond
    for (let i = n - 2; i >= 0; i--) {
        pattern += ' '.repeat(n - i - 1) + '*'.repeat(2 * i + 1) + '\n';
    }

    console.log(pattern);
}

// Prompt the user to enter a number
rl.question('Enter a number: ', (input) => {
    const n = parseInt(input, 10);

    if (isNaN(n) || n <= 0) {
        console.log('Please enter a valid positive number.');
    } else {
        printDiamond(n);
    }

    rl.close();
});

File: task3/GO/Subtask3.go
package main

import (
	"fmt"
)

func main() {
	var n int

	// Read a number from the user
	fmt.Print("Enter a number: ")
	_, err := fmt.Scan(&n)
	if err != nil {
		fmt.Println("Invalid input:", err)
		return
	}

	// Generate and print the diamond pattern
	printDiamond(n)
}

// Function to print the diamond pattern
func printDiamond(n int) {
	// Upper part of the diamond
	for i := 0; i < n; i++ {
		for j := 0; j < n-i-1; j++ {
			fmt.Print(" ")
		}
		for j := 0; j < 2*i+1; j++ {
			fmt.Print("*")
		}
		fmt.Println()
	}

	// Lower part of the diamond
	for i := n - 2; i >= 0; i-- {
		for j := 0; j < n-i-1; j++ {
			fmt.Print(" ")
		}
		for j := 0; j < 2*i+1; j++ {
			fmt.Print("*")
		}
		fmt.Println()
	}
}
